Source/WebCore/platform/audio/cocoa/AudioSampleDataSource.mm

```diff
@@ -207,7 +210,12 @@
         return false;
     }
 
+    uint64_t seekTo = std::exchange(m_seekTo, NoSeek);
+    if (seekTo != NoSeek)
+        m_readCount = seekTo;
```

Comment:
Do we need m_readCount global variable?

Reply:
yes, while a push will typically be with like 4096 frames at a time, from what I've seen pulls are small (128) and this is why we typically have stale data in wait.

So we want to keep track on where we will start from during the next few pulls. The index of the reads is monotonically increasing.

m_readCount indicates that all date prior that index are stale.